align/compiler/find_constraint.py

```diff
@@ -346,7 +347,10 @@ def process_symmnet(self):
                 self.user_constrained_list.append(v.net1)
                 self.user_constrained_list.append(v.net2)
             for symb in new_symmblock_const:
-                self.iconst.append(symb)
+                #avoid subsets
```

Comment:
1. The change below could be refactored to improve readability and to make sure parenthesis is correct. Should not `any()` be applied to an iterable such as `list`? Does not it require `any( [ condition for elem in iterable ] )`?
2. What if the new constraint is not a subset but overlaps with an existing one? Is it preferable to merge rather than add another one? (I thought placer does not allow a block to be on multiple lines of symmetry).

Reply:
1. It works without a separate typecasting it to a list e.g.,
 ```>>> any(i for i in [0,0,0])
False
>>> any(i for i in [0,0,1])
True```
2. Placer combines multiple defined symmetric_block and symmetric_block auto-generated from symmetric_net and creates a superset symmetric_group. During the last discussion, we decided to make the auto-generation of symmetric_block from symmetric_net explicit for the checker. The generated constraint from the symmetric_net constraint are always a single pair so it will always be a subset.

Reply:
Thank you for the clarification.